.github/workflows/caching-build.yml

```diff
@@ -1,14 +1,35 @@
-name: caching-build
+name: build-and-test
+
 on:
   # only runs on main, hourly cache update used by all branches
     schedule:
       - cron: "0 * * * *"
     push:
-
+    pull_request:
+      
 jobs:
+  lint:
+    runs-on: ubuntu-24.04
+    steps:
+      # prevent cache permission errors
+      - run: sudo chown root /usr/bin/tar && sudo chmod u+s /usr/bin/tar      
+      - name: Install deps
+        run: |
+          curl https://sh.rustup.rs -sSf | RUSTUP_INIT_SKIP_PATH_CHECK=yes sh -s -- -y
+          rustup install nightly-2024-09-10
+          export PATH="~/.cargo/bin:$PATH"
+
+      - uses: actions/checkout@v4
+
+      # Lint code
+      - run: cargo +nightly-2024-09-10 fmt
+      - run: git diff --exit-code
+
   build-kernel:
     runs-on: ubuntu-24.04
     steps:
+      # prevent cache permission errors
```

Comment:
Why is this needed everywhere? Is the base image bad?

Reply:
> Why is this needed everywhere? Is the base image bad?

Ehh, bad is a strong term. I'd argue the cache action is not very good for leaving that foot-gun in (everywhere cuz someone will get it wrong in the future otherwise).